RadixWallet/MIGRATE_TO_SARGON/Stage2/Atoms/Stage2MigrateToSargon+ProfileNetwork.swift

```diff
@@ -32,81 +32,104 @@ extension ProfileNetwork {
+		var identified = accounts.asIdentified()
 		for id in idsOfAccountsToHide {
-			accounts[id: id]?.hide()
+			identified[id: id]?.hide()
 			authorizedDapps.mutateAll { dapp in
 				dapp.referencesToAuthorizedPersonas.mutateAll { persona in
 					persona.sharedAccounts?.ids.removeAll(where: { $0 == id })
 				}
 			}
 		}
+		accounts = identified.elements
 	}
 
 	public func getPersonas() -> Personas {
-		personas.nonHidden
+		personas.asIdentified().nonHidden
 	}
 
 	public func getHiddenPersonas() -> Personas {
-		personas.hiden
+		personas.asIdentified().hidden
 	}
 
 	public func personasIncludingHidden() -> Personas {
-		personas
+		personas.asIdentified()
 	}
 
 	public func hasSomePersona() -> Bool {
 		!personas.isEmpty
 	}
 
 	public mutating func addPersona(_ persona: Persona) throws {
-		guard personas.get(id: persona.id) == nil else {
+		var identifiedPersonas = personas.asIdentified()
+		guard identifiedPersonas[id: persona.id] == nil else {
 			throw PersonaAlreadyExists()
 		}
 
-		personas.append(persona)
+		identifiedPersonas.append(persona)
+		self.personas = identifiedPersonas.elements
 	}
 
 	public mutating func updatePersona(_ persona: Persona) throws {
-		guard personas.updateOrAppend(persona) != nil else {
+		var identifiedPersonas = personas.asIdentified()
+		guard identifiedPersonas.updateOrAppend(persona) != nil else {
 			throw TryingToUpdateAPersonaWhichIsNotAlreadySaved()
 		}
+		self.personas = identifiedPersonas.elements
 	}
 
 	public mutating func hidePersonas(ids idsOfPersonaToHide: Set<Persona.ID>) {
+		var identifiedPersonas = personas.asIdentified()
+		var identifiedAuthorizedDapps = authorizedDapps.asIdentified()
 		for id in idsOfPersonaToHide {
 			/// Hide the personas themselves
-			personas[id: id]?.hide()
+			identifiedPersonas[id: id]?.hide()
 
 			/// Remove the persona reference on any authorized dapp
-			authorizedDapps.mutateAll { dapp in
-				dapp.referencesToAuthorizedPersonas.remove(id)
+			identifiedAuthorizedDapps.mutateAll { dapp in
+				var referencesToAuthorizedPersonas = dapp.referencesToAuthorizedPersonas.asIdentified()
+				referencesToAuthorizedPersonas.remove(id: id)
+				dapp.referencesToAuthorizedPersonas = referencesToAuthorizedPersonas.elements
 			}
 		}
+		self.personas = identifiedPersonas.elements
 
 		/// Filter out dapps that do not reference any persona
-		authorizedDapps.filterInPlace(not(\.referencesToAuthorizedPersonas.isEmpty))
+		identifiedAuthorizedDapps.filterInPlace(not(\.referencesToAuthorizedPersonas.isEmpty))
+		self.authorizedDapps = identifiedAuthorizedDapps.elements
 	}
 
 	public mutating func unhideAllEntities() {
-		accounts.mutateAll { $0.unhide() }
-		personas.mutateAll { $0.unhide() }
+		var identifiedAccounts = accounts.asIdentified()
+		var identifiedPersonas = personas.asIdentified()
+		identifiedAccounts.mutateAll { $0.unhide() }
+		identifiedPersonas.mutateAll { $0.unhide() }
```

Comment:
Do we need to get identified array, modify, and get elements on this case? We are not appending/removing elements

Reply:
this protects us if we have a bug in `mutateAll` :) felt safest.